compiler/optimizer/Structure.cpp

```diff
@@ -1223,8 +1231,11 @@ void TR_RegionStructure::collapseIntoParent()
                      if ((*pred)->getFrom()->getNumber() == getNumber() &&
                          toStructureSubGraphNode((*pred)->getFrom())->getStructure() == this)
                         {
-                        nodeInParent->getSuccessors().remove(*pred);
-                        target->getPredecessors().remove(*pred);
+                        TR::CFGEdgeList &successorList = nodeInParent->getSuccessors();
+                        auto successorIterator = std::find(successorList.begin(), successorList.end(), *pred);
+                        if (successorIterator != successorList.end()) successorList.erase(successorIterator);
+
+                        target->getPredecessors().erase(pred);
```

Comment:
Is the find->erase just not required on the getPredecessors case because you know it was found in the successor's list? if so, that might be worthy of a comment in these places where you skip doing the find in the predecessor lists. Or (wait for it) maybe one generic service on cfgnode could take care of it without mostly repeating the code on the client side :) ? Maybe I'm missing something...